src/NuGet.CommandLine/Common/ExitCodeException.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace NuGet.CommandLine
+{
+    [Serializable]
+    public class ExitCodeException : CommandLineException
```

Comment:
The errors are already displayed at this point, there is no message to show in the exception unless we add a restore summary feature.

This extends CommandLineException, it's purpose is for scenarios where this is no message, just a non-zero exit code. Currently a failure here displays a non-user friendly message saying  that a commandlinexception was thrown. In several cases already this has caused users to believe that nuget.exe crashed, which they reported instead of looking at the other error messages to notice that their package sources were incorrect.